js/chat_app/src/redux/chat/slice.js

```diff
@@ -0,0 +1,99 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { chatServices } from '../../services/chatServices';
+
+const initialState = {
+  chatData: null,
+  selectedMembers: [],
+  status: 'fullfield',
+  isError: false
+};
+
+const chatSlice = createSlice({
+  name: 'chat',
+  initialState,
+  reducers: {
+    setSelectedMembers: (state, action) => {
+      state.selectedMembers = action.payload;
+    }
+  },
+  extraReducers: (builder) => {
+    builder
+      .addCase(createChatThunk.fulfilled, (state, action) => {
+        state.status = action.meta.requestStatus;
+        state.isError = false;
+      })
+      .addCase(createChatThunk.pending, (state, action) => {
+        state.status = action.meta.requestStatus;
+      })
+      .addCase(createChatThunk.rejected, (state, action) => {
+        state.status = action.meta.requestStatus;
+        state.isError = true;
+      })
+
+      .addCase(getChatByIdThunk.fulfilled, (state, action) => {
+        state.isError = false;
+        state.status = action.meta.requestStatus;
+        state.chatData = action.payload;
+      })
+      .addCase(getChatByIdThunk.rejected, (state, action) => {
+        state.isError = true;
+        state.status = action.meta.requestStatus;
+      });
+  }
+});
+
+export const addChatMembersThunk = createAsyncThunk(
+  'addMembers/api/chat/:chatId/members',
+  async (paramsForAddMembers, { getState, dispatch }) => {
+    const { authToken } = getState().auth.auth;
+    const { chatId, members } = paramsForAddMembers;
+    const response = await chatServices.addChatMembers(chatId, authToken, members);
+    dispatch(getChatByIdThunk(response?.id, authToken));
+  }
+);
+
+export const createChatThunk = createAsyncThunk(
+  'createChat/api/chats',
+  async (title, { getState, dispatch, rejectWithValue }) => {
+    try {
+      const { authToken } = getState().auth.auth;
```

Comment:
use selectors there too

Reply:
getState() doesn't work with selectors for useSelector()
getState() gives us access to the shared storage, after which we can iterate over the individual state keys and get data from them;
![image](https://user-images.githubusercontent.com/70951729/200384257-eaacd446-d862-4754-a930-87590bce0775.png)

Reply:
so what? :) You can do something like

`const getAuth = (state) => state.auth.auth`;

And call it like
`getAuth(getState())`